sqsdomain/pools.go

```diff
@@ -36,6 +36,15 @@ type PoolI interface {
 	// If this is not a concentrated pool, errors
 	SetTickModel(*TickModel) error
 
+	// GetCWPoolModel returns the CosmWasm pool model for the pool
+	// if this is a CosmWasm pool. Errors otherwise
+	// No error if this is a CosmWasm pool but the model is not set
+	GetCWPoolModel() (*CWPoolModel, error)
```

Comment:
nit: to make it clearer for any future new contributors
```suggestion
	GetCosmWasmPoolModel() (*CWPoolModel, error)
```